python/src/aiconfig/default_parsers/dalle.py

```diff
@@ -0,0 +1,197 @@
+import copy
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+# Type hint imports
+from aiconfig import (
+    ExecuteResult,
+    Output,
+    Prompt,
+    PromptMetadata,
+)
+
+# ModelParser Utils
+from aiconfig import ParameterizedModelParser
+from aiconfig import get_api_key_from_environment
+from aiconfig import resolve_prompt
+from aiconfig import InferenceOptions
+from openai import OpenAI
+
+# Dall-E API imports
+from openai.types import ImagesResponse, Image
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+
+# TODO: Centralize this into utils function with the HuggingFaceTextParser class
+def refine_image_completion_params(model_settings):
+    """
+    Refines the completion params for the Dall-E request API. Removes any unsupported params.
+    The supported keys were found by looking at the OpenAI Dall-E API: https://platform.openai.com/docs/api-reference/images/create?lang=python`
+    """
+    supported_keys = {
+        "model",
+        "n",
+        "quality",
+        "response_format",
+        "size",
+        "style"
+    }
+
+    completion_data = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            completion_data[key.lower()] = model_settings[key]
+
+    return completion_data
+
+
+def construct_regular_output(image_data: Image, execution_count: int) -> Output:
+    output = ExecuteResult(
+        **{
+            "output_type": "execute_result",
+            "data": image_data.b64_json or image_data.url,
+            "execution_count": execution_count,
+            "metadata": {"revised_prompt": image_data.revised_prompt},
+        }
+    )
+    return output
+
+
+class DallE3ImageGenerationParser(ParameterizedModelParser):
+    """
+    A model parser for Dall-E 3 text-->image generation models.
+    """
+    
+    def __init__(self):
+        """
+        Usage:
+
+        1. Create a new model parser object
+                parser = DallE3ImageGenerationParser()
+        2. Add the model parser to the registry.
+                config.register_model_parser(parser)
+
+        The model parser will require an API token to be set in the environment variable OPENAI_API_KEY.
+        """
+        super().__init__()
+        self.client = OpenAI(api_key=get_api_key_from_environment("OPENAI_API_KEY"))
+
+    def id(self) -> str:
+        """
+        Returns an identifier for the model (e.g. dall-e-2, dall-e-3, etc.).
+        """
+        return "dall-e-3"
+
+
+    # TODO: Test this sometime from getting a response from the API and saving it to a file
```

Comment:
@LastMileNewGrad I was planning just to test by calling `DallE3ImageGenerationParser.serialize()` but do you have any other suggestions? I could also directly move the serialize logic from `run_inference()` into here too (I think that's more intuitive)

Reply:
ya you should be able to test by calling serialize, only the run step requires mocking. There should be an example of this test for the openai model parser.

>could also directly move the serialize logic from run_inference() into here too 

Could you clarify what this entails?

Reply:
> Could you clarify what this entails?

In run_inference() we generally have:

1. Building completion params with `deserialize()`
2. Make API request
3. Parse API response into Prompt Output (this step instead of manually parsing, we could call `serialize()` like we do for `deserialize()` in step 1)

Reply:
I see where you're getting at, I think that could be a pretty near api design improvement.

Currently the api for `serialize()` is designed to take completion params and return a `Prompt` Object.

The implicit "serialize" in the run step takes the completion and constructs an Output for a Prompt Object. How would you approach merging the two?

Reply:
Added tests for serialize: https://github.com/lastmile-ai/aiconfig/pull/294